src/libcore/slice/mod.rs

```diff
@@ -62,7 +62,9 @@ impl<T> [T] {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline]
-    #[rustc_const_unstable(feature = "const_slice_len")]
+    #[cfg_attr(boostrap_stdarch_ignore_this, rustc_const_unstable(feature = "const_slice_len"))]
+    // SAFETY: const safe because we transmute out the length field as a usize (which it must be)
```

Comment:
I wonder if, following https://github.com/rust-lang/unsafe-code-guidelines/pull/190, we should call this "const sound" instead?